docker/Dockerfile.backend

```diff
@@ -1,7 +1,11 @@
 ARG BASE_IMAGE=python:3.12-slim
 FROM $BASE_IMAGE AS base
 
-RUN pip install --no-cache-dir --timeout 3 --retries 3 poetry \
+RUN apt-get update && apt-get install -y curl \
+    && rm -rf /var/lib/apt/lists/*
+
+RUN curl -sSL https://install.python-poetry.org | python3 - \
```

Comment:
What's the issue with installing poetry via pip?

Reply:
according to [poetry documentation](https://python-poetry.org/docs/#installing-with-the-official-installer):
> `Poetry should always be installed in a dedicated virtual environment to isolate it from the rest of your system. It should in no case be installed in the environment of the project that is to be managed by Poetry. This ensures that Poetry’s own dependencies will not be accidentally upgraded or uninstalled.`

when installing via pip, it doesn't install poetry dependencies and it was falling on:

`ModuleNotFoundError: No module named 'requests_toolbelt'
`

on line: `RUN poetry install --no-root --all-extras --with test --without docs,dev`

Reply:
maybe but why installing poetry as mention in official poetry docs is worse that installing via pip (poetry documentation doesn't provide installation instruction via pip)?

Reply:
That's just a question "what is wrong with current approach?". I haven't seen any issues with installing poetry via pip before. Maybe something have changed...